src/NUnitFramework/framework/Constraints/CollectionEquivalentConstraintResult.cs

```diff
@@ -0,0 +1,82 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>Provides a <see cref="ConstraintResult"/> for the <see cref="CollectionEquivalentConstraint"/>.</summary>
+    public class CollectionEquivalentConstraintResult : ConstraintResult
+    {
+        /// <summary>Result of a <see cref="CollectionTally"/> of the collections to compare for equivalence.</summary>
+        private CollectionTally.CollectionTallyResult _tallyResult;
```

Comment:
Additionally, the reason I kept a reference to `_lastPerformedTally` was because of how the overrides for `CollectionItemsEqualConstraint` works.  The `Matches` method is called within `ApplyTo` method, and I wanted to avoid performing the tally between the two collections for efficiency reasons.

I've since refactored out a `CollectionTally.CollectionTallyResult` object to hold onto the result of a `CollectionTally`'s processing, so the variable type would need to change accordingly.